modules/gwlb/variables.tf

```diff
@@ -80,21 +157,47 @@ variable "backends" {
       }
     }
   }
-  type = map(any)
+  nullable = false
+  type = map(object({
+    name = optional(string)
+    tunnel_interfaces = map(object({
+      identifier = number
+      port       = number
+      protocol   = optional(string, "VXLAN")
+      type       = string
+    }))
+  }))
+  validation { # protocol
+    condition = (var.backends == null ?
+      true : alltrue(flatten([for k, v in var.backends :
+    [for p, r in v.tunnel_interfaces : contains(["VXLAN"], r.protocol)]])))
+    error_message = "The `protocol` property can be only \"VXLAN\"."
+  }
+  validation { # type
+    condition = (var.backends == null ?
+      true : alltrue(flatten([for k, v in var.backends :
+    [for p, r in v.tunnel_interfaces : contains(["Internal", "External"], r.type)]])))
+    error_message = "The `type` property can be one of \"Internal\", \"External\"."
+  }
 }
 
 variable "lb_rule" {
   description = <<-EOF
-  Load balancing rule config. Available options:
-  - `name`              - (Optional|string) Name for the rule. Defaults to `var.frontend_ip_config.name`.
-  - `load_distribution` - (Optional|string) Refer to [provider docs](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/lb_rule#load_distribution).
-  EOF
-  default     = null
-  type        = map(string)
-}
+  Load balancing rule configuration.
 
-variable "tags" {
-  description = "Azure tags to apply to the created resources."
+  Available options:
+  - `name`              - (`string`, optional) name for the rule.
+  - `load_distribution` - (`string`, optional, defaults to `Default`) specifies the load balancing distribution type
+                          to be used by the Gateway Load Balancer.
+  EOF
   default     = {}
-  type        = map(string)
+  nullable    = false
+  type = object({
```

Comment:
We cannot use `optional` for object - here is the error:

```
Invalid type specification: Keyword "optional" is valid only as a modifier for object type attributes.
```

Frankly speaking I've done it by purpose as we have always 1 rule , which is required , so I propose to leave it as it is - do you agree?

Reply:
So you are able to create a rule without a name, correct?

Reply:
no, you cannot, according to provider's documentation, `name` [is required](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/lb_rule#name)

if we want to have this as optional i would suggest something like this

```hcl
default = null
nullable = true
type = object({
  name = string
  load_distribution = optional(string, "Default")
})
```

This would require minor changes in the code to handle `null`

Reply:
no, I was not creating a rule without name - in case name was not provided, I used:

`name = coalesce(var.lb_rule.name, azurerm_lb.this.frontend_ip_configuration[0].name)`

Nevertheless I'm going to change as required to have the same approach for naming as for other resources.